### PR TITLE
Add cpp check

### PR DESCRIPTION
**Pull request by jiangpeipei327**

### CODE REVIEW

File: internal/linters/util.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/qiniu/x/log"
+	"github.com/qiniu/x/xlog"
+)
+
+func IsEmpty(args ...string) bool {
```
[the remark this answers is not in the document]
Comment (jiangpeipei327) [reply internal/linters/util.go]: fix

File: internal/linters/util.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	return &LinterOutput{
+		File:    matches[1],
+		Line:    int(lineNumber),
+		Column:  int(columnNumber),
+		Message: matches[4],
+	}, nil
+}
+
+var gop_auto_generated_file_pattern = `^.*_autogen.*.go$`
+var gopGeneratedFileRegexp = regexp.MustCompile(gop_auto_generated_file_pattern)
```
[the remark this answers is not in the document]
Comment (jiangpeipei327) [reply internal/linters/util.go]: fix

File: internal/linters/util.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+func isGopAutoGeneratedFile(file string) bool {
```
[the remark this answers is not in the document]
Comment (jiangpeipei327) [reply internal/linters/util.go]: fix

File: internal/linters/c/cppcheck/cppcheck.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+	if linters.IsEmpty(a.LinterConfig.Args...) {
+		a.LinterConfig.Args = append([]string{}, "--quiet", "--template='{file}:{line}:{column}: {message}'", "./")
+	}
+
+	return linters.GeneralHandler(log, a, linters.GeneralParse)
+}
+
+// ExecRun executes a command.
+func cppCheckExecRun(workDir, command string, args ...string) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (jiangpeipei327) [reply internal/linters/c/cppcheck/cppcheck.go:27]: fix

[the remark this answers is not in the document]
Comment (jiangpeipei327) [reply internal/linters/c/cppcheck/cppcheck.go:27]: fix

File: internal/linters/c/cppcheck/cppcheck.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package cppcheck
+
+import (
+	"github.com/qiniu/reviewbot/internal/linters"
+	"github.com/qiniu/x/xlog"
+)
+
+var linterName = "cppcheck"
```
Comment (CarlJi) [inline internal/linters/c/cppcheck/cppcheck.go:8]: add official reference for this tool

File: config/config.yaml [hunk 6]
```diff
@@ -26,4 +26,9 @@ qbox/kodo: # github repository name, which will override the settings in qbox or
     workDir: "src/qiniu.com/kodo"
     command: staticcheck
     args: ["-debug.no-compile-errors=true", "./..."]
-    run_if_changed: ["src/qiniu.com/kodo/.*\\.go$"] # only run if changed files match the regex
\ No newline at end of file
+    run_if_changed: ["src/qiniu.com/kodo/.*\\.go$"] # only run if changed files match the regex
+
+qbox/3rd-ffmpeg: # github repository name, which will override the settings in qbox org
```
Comment (CarlJi) [inline config/config.yaml:31]: why do this config? since no custom configs

File: internal/linters/c/cppcheck/cppcheck.go [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+func init() {
+	linters.RegisterPullRequestHandler(linterName, cppcheckHandler)
+}
+
+func cppcheckHandler(log *xlog.Logger, a linters.Agent) error {
+
+	if linters.IsEmpty(a.LinterConfig.Args...) {
```
Comment (CarlJi) [inline internal/linters/c/cppcheck/cppcheck.go:16]: recommend to add a readme to introduce it.

File: config/config.yaml (hunk 6 above)
Comment (jiangpeipei327) [reply config/config.yaml:31]: https://github.com/qbox/3rd-ffmpeg

File: internal/linters/c/cppcheck/cppcheck.go (hunk 5 above)
Comment (jiangpeipei327) [reply internal/linters/c/cppcheck/cppcheck.go:8]: done

File: internal/linters/c/cppcheck/cppcheck.go (hunk 7 above)
Comment (jiangpeipei327) [reply internal/linters/c/cppcheck/cppcheck.go:17]: done

File: Dockerfile [hunk 8]
```diff
@@ -24,6 +24,8 @@ RUN apt-get update && apt-get install -y luarocks \
     && luarocks install luacheck \
     && rm -rf /var/lib/apt/lists/*
 
+# install cppcheck lint tools
+RUN apt-get update && apt-get install -y cppcheck
```
[the remark this answers is not in the document]
Comment (jiangpeipei327) [reply Dockerfile:28]: fix

**Pull request merged**